### PR TITLE
Implement JsonUtils

### LINKED ISSUE #22768: [Task]: Implement JsonUtils
### What needs to happen?

The output of this task is to create `org.apache.beam.sdk.schemas.utils.JsonUtils`.

The purpose is to have a central place for helper methods to convert to and from Json encoding in the setting of schema-aware PCollections.

The design will pattern similar but not exactly as AvroUtils in the same package.

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-java-core

### PR DESCRIPTION
**Pull request by damondouglas**
This PR fixes #22768 by implementing JsonUtils with SimpleFunctions.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - ~Update `CHANGES.md` with noteworthy changes.~
 - ~If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).~

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damondouglas** [comment]
@pabloem FYI what we chatted about.

**Comment by damondouglas** [comment]
R: @pabloem

**Comment by pabloem** [comment]
so sorry about the delay. Looking now.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/utils/JsonUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        JAVA_BEAN_SCHEMA.toRowFunction(TYPE_DESCRIPTOR);
+
+    static Cat of(String name, double weight, int age, boolean spayNeutered) {
+      return new Cat(name, weight, age, spayNeutered);
+    }
+
+    private final String name;
+    private final double weight;
+    private final int age;
+    private final boolean spayNeutered;
```
Comment (pabloem) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/utils/JsonUtilsTest.java:125]: I wonder if it makes sensse to have other types (float, long, DateTime)

**Comment by pabloem** [comment]
@damondouglas I'd still like to follow up on the extra types. LMK what you think.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/utils/JsonUtilsTest.java (hunk 1 above)
Comment (damondouglas) [reply sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/utils/JsonUtilsTest.java:125]: Good point!  I will add these in an additional PR.

**Pull request merged**